src/common/cmdparse.h

```diff
@@ -79,19 +79,16 @@ bool cmd_getval(const cmdmap_t& cmdmap,
 
 // with default
 
-template <typename T>
-bool cmd_getval(
-  const cmdmap_t& cmdmap, std::string_view k,
-  T& val, const T& defval)
+template <typename T, typename V>
+T cmd_getval(const cmdmap_t& cmdmap, std::string_view k,
+	     const V& defval)
 {
   auto found = cmdmap.find(k);
   if (found == cmdmap.end()) {
-    val = defval;
-    return true;
+    return T(defval);
```

Comment:
i fixed by returning `T(defval)`, instead of `std::forward<V>(defval)`.

Reply:
Huh, do we really had / still have an usage that passed rvalue reference and assumed the callee won't move-out from it? If so, this might be an indicator of another problem.

Reply:
you are right, we don't do this actually. but after a second thought. i think we should not use perfect forwarding here. for two reasons

- we never call this function like `cmd_getval_or(..., std::move(foo))`, so unless we have `cmd_getval_or(..., V& defval)`, we are not allowed to move from `defval`.
- we cannot benefit from constructing the return value by perfect forwarding an argument. unless we have something like
  ```c++
  string val = cmd_getval_or<string>(cmdmap, "val_name", std::string("foo"));
  ```
  but what we have in general is
  ```c++
  string val = cmd_getval_or<string>(cmdmap, "val_name", "foo");
  ```